feat: create component PIN code input

nouha06 opened the pull request:
_Note: Please transform `- [ ]` into `- (NA)` in the description when things are not applicable_

### Related issues

[#307] https://github.com/Orange-OpenSource/ouds-flutter/issues/307 

### Description

<!-- Describe your changes in detail -->

### Motivation & Context

<!-- Why is this change required? What problem does it solve? -->

### Types of change

<!-- What types of changes do your code introduce? -->
<!-- Please remove the unused items in the list -->

- Bug fix (non-breaking which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Refactoring (non-breaking change)
- Breaking change (fix or feature that would change existing functionality)

### Previews

<!-- Please add screenshots or videos showing your evolutions -->

### Checklist

<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!-- Note that any checkboxes not optional must be ticked by an 'x' (or '(NA)') and our [branch ruleset](https://github.com/marketplace/task-list-completed) may block any merge if some mandatory boxes remain empty -->
<!-- Your branch used to submit the evolutions must be prefixed by the issue number like 666-add-some-feature -->

#### Contribution

- [x] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/ouds-flutter/blob/develop/CONTRIBUTING.md)

#### Accessibility

- [ ] My change follows accessibility good practices

#### Design

- [x] My change respects the design guidelines of _Orange Unified Design System_

#### Development

- [x] My change follows the [developer guide](https://github.com/Orange-OpenSource/ouds-flutter/blob/develop/DEVELOP.md)
- [ ] I have added unit tests to cover my changes _(optional)_

#### Documentation

- [x] My change introduces changes to the documentation and/or I have updated the documentation accordingly

### Checklist (for Core Team only)

- [ ] Manually test (dark mode, RTL, landscape display, tablet)
- [x] Documentation has been updated if relevant
- [ ] Design review
- [ ] A11y review
- [ ] Internal files have been updated if relevant (THIRD_PARTY, NOTICE)
- [x] changelog.md has been updated respecting [keep a changelog rules](https://keepachangelog.com/en/1.0.0/) and referencing the issue

nouha06 committed:
feat: prepare the Pin Code Input Component in core

nouha06 committed:
feat: prepare the Pin Code Input Component in core

nouha06 committed:
feat:add token textInput

nouha06 committed:
feat:add token textInput

nouha06 committed:
feat: add textField for code pin input

nouha06 committed:
feat: add textField for code pin input

nouha06 committed:
Merge branch 'develop' into 307-create-component---pin-code-input

# Conflicts:
#	ouds_theme_contract/lib/theme/tokens/components/ouds_components_tokens.dart
#	ouds_theme_orange/lib/orange_theme.dart
#	ouds_theme_sosh/lib/ouds_theme_sosh.dart

nouha06 committed:
chore: add pin code input component in lib

nouha06 committed:
chore: add pin code input component in lib

nouha06 committed:
feat: add pin code input component in app demo

nouha06 committed:
feat: add pin code input component in app demo

nouha06 committed:
fix: delete unused import

nouha06 committed:
fix: delete unused import

nouha06 committed:
feat: fix hint text

nouha06 committed:
feat: fix hint text

nouha06 committed:
feat: add variant pin code input in demo app

nouha06 committed:
feat: add variant pin code input in demo app

nouha06 committed:
feat: add OudsDigitInput

nouha06 committed:
feat: add OudsDigitInput

nouha06 committed:
feat: refactor code and add verification code in demo app

nouha06 committed:
feat: refactor code and add verification code in demo app

nouha06 committed:
feat: refactor

nouha06 committed:
feat: refactor

nouha06 committed:
feat: refactor code and documentation, and add verification state in demo app

nouha06 committed:
feat: refactor code and documentation, and add verification state in demo app

AhmedAmineZr commented:
don't forget to make rebase, we have this issue you need to update : 
`Merge the latest changes from develop into this branch. This merge commit will be associated with AhmedAmineZr.`

nouha06 committed:
Merge branch '307-create-component---pin-code-input' of https://github.com/Orange-OpenSource/ouds-flutter into 307-create-component---pin-code-input

nouha06 committed:
review : review code

nouha06 committed:
Merge branch 'develop' into 307-create-component---pin-code-input

nouha06 committed:
feat : update dart doc options

nouha06 committed:
Merge branch 'develop' into 307-create-component---pin-code-input

nouha06 committed:
review : update the documentation of digit input and pin code input

nouha06 committed:
chore: delete the style option from the pin code input component and use only on digit input decoration

nouha06 committed:
review : refactor code for the component and fix some issues in demo app

nouha06 committed:
review : add use pin code input control state instead of text input control state

nouha06 committed:
Merge branch 'develop' into 307-create-component---pin-code-input

# Conflicts:
#	ouds_core/lib/l10n/ouds_flutter_ar.arb

nouha06 committed:
Merge branch 'develop' into 307-create-component---pin-code-input

# Conflicts:
#	app/CHANGELOG.md
#	ouds_core/CHANGELOG.md

nouha06 committed:
review : update helper text value according to the number of digit input selected

nouha06 committed:
Merge branch 'develop' into 307-create-component---pin-code-input

# Conflicts:
#	ouds_core/dartdoc_options.yaml
#	ouds_core/lib/l10n/gen/ouds_localizations.dart
#	ouds_core/lib/l10n/gen/ouds_localizations_ar.dart
#	ouds_core/lib/l10n/gen/ouds_localizations_en.dart
#	ouds_core/lib/l10n/ouds_flutter_ar.arb
#	ouds_core/lib/l10n/ouds_flutter_en.arb

nouha06 committed:
Merge branch 'refs/heads/develop' into 307-create-component---pin-code-input

nouha06 committed:
fix: separate Pin Code Input from generic Input component in component list and fix the duplicate "Hidden Password" entry in customization bottom sheet

nouha06 committed:
fix : digit padding changed according to the length of Pin Code input component

nouha06 committed:
Merge branch 'develop' into 307-create-component---pin-code-input

# Conflicts:
#	ouds_core/lib/l10n/gen/ouds_localizations.dart

nouha06 committed:
review : fix PIN code input error handling

nouha06 committed:
Merge branch 'develop' into 307-create-component---pin-code-input

nouha06 committed:
fix: update accessibility

nouha06 committed:
fix: update accessibility for pin code input

nouha06 committed:
fix: Pin Code Input behavior and component ordering

- Ensure component pages are always listed alphabetically.
- Allow users to directly replace an existing digit without deleting it first.
- Support full copy/paste of PIN
- Fix delete/backspace behavior
- Add documentation URL

nouha06 committed:
Merge branch 'develop' into 307-create-component---pin-code-input

nouha06 committed:
Merge branch 'develop' into 307-create-component---pin-code-input

# Conflicts:
#	app/CHANGELOG.md
#	ouds_core/CHANGELOG.md

nouha06 committed:
review: Ensure component pages are listed in alphabetical order

nouha06 committed:
Merge branch 'develop' into 307-create-component---pin-code-input

# Conflicts:
#	app/CHANGELOG.md
#	app/lib/l10n/gen/ouds_flutter_app_localizations.dart
#	app/lib/l10n/gen/ouds_flutter_app_localizations_ar.dart
#	app/lib/l10n/gen/ouds_flutter_app_localizations_en.dart
#	app/lib/l10n/ouds_flutter_ar.arb
#	app/lib/l10n/ouds_flutter_en.arb
#	app/lib/ui/components/components.dart
#	app/lib/ui/utilities/customizable/customizable_textfield.dart
#	ouds_core/CHANGELOG.md

nouha06 committed:
review: allow replacing existing digit without deleting first

Tayebsed93 committed:
Merge branch 'develop' into 307-create-component---pin-code-input

nouha06 committed:
Merge branch 'develop' into 307-create-component---pin-code-input

nouha06 committed:
Merge remote-tracking branch 'origin/307-create-component---pin-code-input' into 307-create-component---pin-code-input
